src/DotNetLightning.Core/Utils/Extensions.fs

```diff
@@ -101,11 +101,34 @@ type BitArrayExtensions() =
 type System.Collections.BitArray with
     member this.ToByteArray() =
```

Comment:
The method has become quite lengthy, can you add a test to assure that it is actually the inverse of `FromBytes` somehow?
It can be either property based test or regular unit test. And you can test the behavior via `FeatureBit` instead of testing this method directly if that sounds better.

Reply:
Okay, I've modified the property test that was there and added an extra unit test.